client/rest-high-level/src/test/java/org/elasticsearch/client/RestHighLevelClientTests.java

```diff
@@ -320,6 +326,50 @@ public void testParseEntity() throws IOException {
         }
     }
 
+    public void testParseCompressedEntity() throws IOException {
+        CheckedFunction<XContentParser, String, IOException> entityParser = parser -> {
+            assertEquals(XContentParser.Token.START_OBJECT, parser.nextToken());
+            assertEquals(XContentParser.Token.FIELD_NAME, parser.nextToken());
+            assertTrue(parser.nextToken().isValue());
+            String value = parser.text();
+            assertEquals(XContentParser.Token.END_OBJECT, parser.nextToken());
+            return value;
+        };
+
+        HttpEntity jsonEntity = new NByteArrayEntity(compress("{\"field\":\"value\"}"), ContentType.APPLICATION_JSON);
+        assertEquals("value", restHighLevelClient.parseEntity(enrichHeaderContentEncodingWithGzip(jsonEntity), entityParser));
+        HttpEntity yamlEntity = new NByteArrayEntity(compress("---\nfield: value\n"), ContentType.create("application/yaml"));
+        assertEquals("value", restHighLevelClient.parseEntity(enrichHeaderContentEncodingWithGzip(yamlEntity), entityParser));
+        HttpEntity smileEntity = createCompressedBinaryEntity(SmileXContent.contentBuilder(), ContentType.create("application/smile"));
+        assertEquals("value", restHighLevelClient.parseEntity(enrichHeaderContentEncodingWithGzip(smileEntity), entityParser));
+        HttpEntity cborEntity = createCompressedBinaryEntity(CborXContent.contentBuilder(), ContentType.create("application/cbor"));
+        assertEquals("value", restHighLevelClient.parseEntity(enrichHeaderContentEncodingWithGzip(cborEntity), entityParser));
+    }
+
+    private static byte[] compress(String content) throws IOException {
+        return compress(content.getBytes(StandardCharsets.ISO_8859_1));
+    }
+
+    private static byte[] compress(byte[] content) throws IOException {
+        ByteArrayOutputStream bos = new ByteArrayOutputStream(content.length);
+        GZIPOutputStream gzip = new GZIPOutputStream(bos);
+        gzip.write(content);
+        gzip.close();
+        bos.close();
+
+        return bos.toByteArray();
+    }
+
+    private static HttpEntity enrichHeaderContentEncodingWithGzip(HttpEntity httpEntity) {
```

Comment:
Yes, also done. And instead of using BasicHttpEntity I have used NByteArrayEntity and included the content encoding and discarded enrichHeaderContentEncodingWithGzip method as that one is not needed anymore. Thank you for the feedback!